pkg/handlers/v1/nexposevulnnotifier_test.go

```diff
@@ -43,5 +45,142 @@ func TestNexposeVulnNotificationHandler(t *testing.T) {
 		SiteID: "12345",
 	}
 	handler.Handle(ctx, scanInfo)
+}
+
+func TestNexposeVulnNotificationHandlerMultipleAssets(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	assetFetcher := NewMockAssetFetcher(mockCtrl)
+	producer := NewMockProducer(mockCtrl)
+
+	assetChan := make(chan domain.AssetEvent, 2)
+	errChan := make(chan error, 1)
+
+	asset := domain.AssetEvent{ID: 12345}
+	asset2 := domain.AssetEvent{ID: 56789}
+
+	assetChan <- asset
+	assetChan <- asset2
+	close(assetChan)
+	close(errChan)
+
+	assetFetcher.EXPECT().FetchAssets(gomock.Any(), "12345").Return(assetChan, errChan)
+	producer.EXPECT().Produce(gomock.Any(), gomock.Any()).Return(nil).Times(2)
+
+	handler := NexposeVulnNotificationHandler{
+		Producer:     producer,
+		AssetFetcher: assetFetcher,
+		LogFn:        runhttp.LoggerFromContext,
+		StatFn:       runhttp.StatFromContext,
+	}
+
+	ctx := logevent.NewContext(context.Background(), logevent.New(logevent.Config{Output: ioutil.Discard}))
+	scanInfo := ScanInfo{
+		SiteID: "12345",
+	}
+	handler.Handle(ctx, scanInfo)
+}
 
+func TestNexposeVulnNotificationHandlerError(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	assetFetcher := NewMockAssetFetcher(mockCtrl)
+	producer := NewMockProducer(mockCtrl)
+
+	assetChan := make(chan domain.AssetEvent, 1)
+	errChan := make(chan error, 1)
+
+	errChan <- errors.New("myError")
+	close(assetChan)
+	close(errChan)
+
+	assetFetcher.EXPECT().FetchAssets(gomock.Any(), "12345").Return(assetChan, errChan)
+	producer.EXPECT().Produce(gomock.Any(), gomock.Any()).Return(nil).Times(0)
+
+	handler := NexposeVulnNotificationHandler{
+		Producer:     producer,
+		AssetFetcher: assetFetcher,
+		LogFn:        runhttp.LoggerFromContext,
```

Comment:
we should create a ticket to remove `runhttp` from tests

Reply:
Why? What should be used instead?

Reply:
There was a lot of discussion in this [pull request](https://github.com/asecurityteam/nexpose-asset-attributor/pull/1), but I think this [comment](https://github.com/asecurityteam/nexpose-asset-attributor/pull/1#discussion_r278679561) sums it up best.

Reply:
I see. Thanks for the call-out. Made ticket SECD-539 to track that work